src/vm/vm_memory/memory.rs

```diff
@@ -300,6 +301,29 @@ impl Memory {
     }
 }
 
+impl Display for Memory {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        writeln!(f, "temp_memory {{")?;
```

Comment:
We could avoid making distinctions between real and temporary memory and just pritnt them out as a single memory. Temporary segments can be identified by their negative index